docs/kb/semgrep-ci/jenkins-shared-library-with-semgrep.md

```diff
@@ -0,0 +1,73 @@
+---
+tags:
+  - Jenkins
+  - Shared Library
+description: Setting up Jenkins shared library with Semgrep scans
+---
+
+# Jenkins shared library with Semgrep scans
+
+## Motivation
+A good practice when programming is to avoid duplicate code. This applies even when defining your build pipelines. To prevent duplicate code when defining Semgrep tests with Jenkins, create a [Jenkins Shared Library](https://www.jenkins.io/doc/book/pipeline/shared-libraries/) with the Semgrep snippet. Then, call this library in all the projects you have in your organization.
+
+## Creating a Jenkins pipeline to call Semgrep
+If you follow [Semgrep documentation](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-jenkins-configuration-snippet), you can create a simple Jenkins pipeline to run Semgrep scans.
+
+When rolling out Semgrep in your organization, you need to replicate this pipeline to the hundreds of projects you have in your company. 
+After some days, you got it, but then your manager asks you to generate json reports after every semgrep scan to dump results in DefectDojo. It means adding some flags to the semgrep command:
+`````
+semgrep ci --json --output output.json
+`````
+But it must be done in all the repositories you have! It will take you time!
+
+## Creating a shared library
+Jenkins shared library comes to the rescue. Basically, you encapsulate the semgrep commands in a common library.
+The steps are:
+
+1. Create a new repo with a groovy file and add the semgrep snippet there:
+
+`````
+def call() {
+  sh '''docker pull returntocorp/semgrep && \
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -e SEMGREP_REPO_NAME=$SEMGREP_REPO_NAME \
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            returntocorp/semgrep semgrep ci '''
+}
+`````
+> Note: this file must be in a folder called “vars” example: vars/semgrepFullScan.groovy
```

Comment:
I might move this note up to the instruction in step 1:

Create a new repo with a groovy file in the `vars` folder. For example, `vars/semgrepFullScan.groovy`. Add the Semgrep snippet to that file: